pkg/controlplane/server/http/authz.go

```diff
@@ -43,7 +43,13 @@ func (s *Server) PeerAuthorize(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	resp, err := s.cp.AuthorizeIngress(&controlplane.IngressAuthorizationRequest{Service: req.Service})
+	peer, err := s.cp.AuthorizePeer(r.TLS.PeerCertificates[0].DNSNames)
```

Comment:
The purpose of the `Server` struct is just parsing HTTP and passing it on to the lower layer (which is `controlplane::Instance`).
Thus, my view is that the new `AuthorizePeer` function should be merged to `AuthorizeIngress`.